tee-worker/litentry/core/assertion-build/src/a14.rs

```diff
@@ -113,26 +50,28 @@ pub fn build(
 		}
 	}
 	let mut value = false;
-	let mut client = A14Client::new(data_provider_config);
-
-	for address in polkadot_addresses {
-		let data = A14Data {
-			params: A14DataParams { address },
-			include_metadata: false,
-			include_widgets: false,
-		};
-		let response = client.send_request(&data)?;
-
-		let result = response
-			.data
-			.get("result")
-			.and_then(|r| r.as_bool())
-			.ok_or(Error::RequestVCFailed(Assertion::A14, ErrorDetail::ParseError))?;
-		if result {
-			value = result;
-			break
-		}
-	}
+	let mut client = AchainableClient::new(data_provider_config);
+
+	loop_with_abort_strategy(
+		polkadot_addresses,
+		|address| {
+			let data = LabelQueryReq {
+				params: LabelQueryReqParams { address: address.clone() },
+				include_metadata: false,
+				include_widgets: false,
+			};
+			let result = client
+				.query_label("a719e99c-1f9b-432e-8f1d-cb3de0f14dde", &data)
+				.map_err(|e| Error::RequestVCFailed(Assertion::A14, e.into_error_detail()))?;
+			if result {
+				value = result;
+				Ok(LoopControls::Break)
+			} else {
+				Ok(LoopControls::Continue)
+			}
+		},
+		AbortStrategy::ContinueUntilEnd::<fn(&_) -> bool>,
+	)?;
```

Comment:
Hey @kziemianek see here:

response 1 gets a network error
code goes to else block, i.e. continue
response 2 gets a false value
code goes to else block, i.e. continue
response 3 gets a true value
code goes to if branch, i.e. break

So any network error will just go to else block, won't break the loop. Does this make more sense?

Reply:
Also
<img width="770" alt="Screenshot 2024-05-10 at 7 47 01 pm" src="https://github.com/litentry/litentry-parachain/assets/120463031/351aed1e-e69f-4dab-bc57-bbba36aced79">

Once AbortStrategy::ContinueUntilEnd is used in above L73, if not the last element then it won't break the loop.

Reply:
It makes same sense as before 🙃. Honestly, I'm not sure which version is better (previous or current) but it seems like the previous was more accurate.


(1) network error (potentailly true)
(2) false
(3) false

result: false